components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -54,6 +54,20 @@ func (s *OIDCService) CreateClientConfig(ctx context.Context, req *connect.Reque
 		return nil, err
 	}
 
+	conn, err := s.getConnection(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	_, userID, err := s.getUser(ctx, conn)
+	if err != nil {
+		return nil, err
+	}
+
+	if authorizationErr := s.userIsOrgOwner(ctx, userID, organizationID); authorizationErr != nil {
+		return nil, authorizationErr
+	}
```

Comment:
After this lands, we can do a pass to merge these blocks together as we nearly always do all of them in this exact order, but I didn't want to pollute this PR with the refactor